include/mfmg/sparse_matrix_device.templates.cuh

```diff
@@ -241,6 +241,37 @@ SparseMatrixDevice<ScalarType>::SparseMatrixDevice(
   ASSERT_CUSPARSE(cusparse_error_code);
 }
 
+template <typename ScalarType>
+void SparseMatrixDevice<ScalarType>::reinit(
+    MPI_Comm comm, ScalarType *val_dev_, int *column_index_dev_,
+    int *row_ptr_dev_, unsigned int local_nnz,
+    dealii::IndexSet const &range_indexset,
+    dealii::IndexSet const &domain_indexset, cusparseHandle_t cusparse_handle_)
+{
+  // This function can only be called if the object is empty. Otherwise we need
+  // to deal with the ownership of the data
```

Comment:
This seems like it will cause problems at some points. Can we assert this?

Reply:
done